tests/SideBySide/ConnectionPool.cs

```diff
@@ -160,6 +160,37 @@ public void ResetConnection(object connectionReset, int poolSize, long expected)
 		}
 
 		[Fact]
+		public async Task ConnectionLifeTime()
+		{
+			var csb = AppConfig.CreateConnectionStringBuilder();
+			csb.Pooling = true;
+			csb.MinimumPoolSize = 0;
+			csb.MaximumPoolSize = 1;
+			csb.ConnectionLifeTime = 3;
+			int serverThread;
+
+			using (var connection = new MySqlConnection(csb.ConnectionString))
+			{
+				await connection.OpenAsync();
+				serverThread = connection.ServerThread;
+				await Task.Delay(TimeSpan.FromSeconds(1));
+			}
+
+			using (var connection = new MySqlConnection(csb.ConnectionString))
```

Comment:
Baseline fails here in AppVeyor: https://ci.appveyor.com/project/mysqlnet/mysqlconnector/build/1.0.389#L423